components/automate-ui/src/app/entities/environments/environment.model.ts

```diff
@@ -1,3 +1,36 @@
 export interface Environment {
   name: string;
+  chef_type: string;
+  description: string;
+  json_class: string;
+  default_attributes: string;
+  override_attributes: string;
+  run_list: string[];
+  cookbook_versions: CookbookVersion[];
+}
+
+export interface CookbookVersion {
+  name: string;
+  version: string;
+}
+
+export class EnvironmentAttributes {
+  default_attributes: string;
+  override_attributes: string;
+  all: Object;
```

Comment:
@msorens I have changed the type of `all` to Object because only we need to default and override attributes as keys also I applied the same changes for the roles.

Reply:
Please examine this block of code--what you changed does _not_ fix it; in a way, it makes it worse.

https://github.com/chef/automate/blob/fcfad120a7077e9c2c14554af6e61252f73472b3/components/automate-ui/src/app/entities/environments/environment.model.ts#L17-L25
The expression `JSON.parse(resp.default_attributes)` returns an Object, **not** a string, so you are storing an object into a variable declared as a string. That tells me that the code is mixing up types and may get unexpected results because of these inconsistencies. 
When I run it in the browser, I saw `resp.default_attributes` come in first as an empty string, so you forced that to set `this.default_attributes` to empty string. _That is the only value where `this.default_attributes` will be set to a string._

Letting the code run further my breakpoint was hit again, this time with  `resp.default_attributes` as `'{}'`, which sets `this.default_attributes` to an empty _object_.
As far as I can see, `default_attributes` and `override_attributes` need to be declared as objects and the code should be reverted to this:
```
this.default_attributes =
      (resp.default_attributes && JSON.parse(resp.default_attributes)) || {};
this.override_attributes =
      (resp.override_attributes && JSON.parse(resp.override_attributes)) || {};
```